src/pages/Main.test.ts

```diff
@@ -0,0 +1,210 @@
+  return {
+    ...jest.requireActual<FirecloudNotificationExports>('src/components/FirecloudNotification'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type IdleStatusMonitorExports = typeof import('src/components/IdleStatusMonitor') & { __esModule: true };
+jest.mock('src/components/IdleStatusMonitor', (): IdleStatusMonitorExports => {
+  return {
+    ...jest.requireActual<IdleStatusMonitorExports>('src/components/IdleStatusMonitor'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type SupportRequestExports = typeof import('src/components/SupportRequest') & { __esModule: true };
+jest.mock('src/components/SupportRequest', (): SupportRequestExports => {
+  return {
+    ...jest.requireActual<SupportRequestExports>('src/components/SupportRequest'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type TitleManagerExports = typeof import('src/components/TitleManager');
+jest.mock('src/components/TitleManager', (): TitleManagerExports => {
+  return {
+    ...jest.requireActual<TitleManagerExports>('src/components/TitleManager'),
+    TitleManager: jest.fn().mockReturnValue(null),
+  };
+});
+
+type ImportStatusExports = typeof import('src/data/ImportStatus') & { __esModule: true };
+jest.mock('src/data/ImportStatus', (): ImportStatusExports => {
+  return {
+    ...jest.requireActual<ImportStatusExports>('src/data/ImportStatus'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type BrandUtilsExports = typeof import('src/libs/brand-utils');
+jest.mock('src/libs/brand-utils', (): BrandUtilsExports => {
+  const { defaultBrand } = jest.requireActual('src/libs/brands');
+  return {
+    ...jest.requireActual<BrandUtilsExports>('src/libs/brand-utils'),
+    getEnabledBrand: jest.fn().mockReturnValue(defaultBrand),
+    isBrand: jest.fn().mockImplementation((brand) => brand === defaultBrand),
+  };
+});
+
+type ErrorExports = typeof import('src/libs/error');
+jest.mock('src/libs/error', (): ErrorExports => {
+  return {
+    ...jest.requireActual<ErrorExports>('src/libs/error'),
+    reportError: jest.fn(),
+  };
+});
+
+type EventsExports = typeof import('src/libs/events');
+jest.mock('src/libs/events', (): EventsExports => {
+  return {
+    ...jest.requireActual<EventsExports>('src/libs/events'),
+    PageViewReporter: jest.fn().mockReturnValue(null),
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock('src/libs/nav', (): NavExports => {
+  return {
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    getCurrentUrl: jest.fn().mockReturnValue(''),
+    LocationProvider: jest.fn().mockImplementation(({ children }) => children),
+    PathHashInserter: jest.fn().mockReturnValue(null),
+    Router: jest.fn().mockReturnValue(null),
+  };
+});
+
+describe('Main', () => {
+  describe('theme', () => {
+    it('renders ThemeProvider with theme from enabled brand', () => {
```

Comment:
Bonus test!

There aren't many places where Main does anything besides render a component. This is the one of the few places where it integrates two pieces: the brand and the ThemeProvider.

Reply:
I love this... a bit annoying how many top-level components needed to be mocked, but now we have a test file to keep this critical root setup stuff well tested.  BRAVO!